File: .sdk-repos/version-8.9/orchestration-cluster-api-python/examples/process_instance.py
# Compilable usage examples for process instance operations.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    CamundaClient,
    LimitBasedPagination,
    ProcessCreationById,
    ProcessCreationByKey,
    ProcessDefinitionId,
    ProcessDefinitionKey,
    ProcessInstanceSearchQuery,
    ProcessInstanceSearchQueryFilter,
    ProcessInstanceSearchQuerySortRequest,
    ProcessInstanceSearchQuerySortRequestField,
    SortOrderEnum,
)


# region CreateProcessInstanceByKey
def create_process_instance_by_key_example() -> None:
    client = CamundaClient()

    # Deploy a process and obtain the typed key from the response
    deployment = client.deploy_resources_from_files(["order-process.bpmn"])
    process_key = deployment.processes[0].process_definition_key

    # Use the typed key directly — no manual string lifting needed
    result = client.create_process_instance(
        data=ProcessCreationByKey(
            process_definition_key=process_key,
        )
    )

    print(f"Process instance key: {result.process_instance_key}")
# endregion CreateProcessInstanceByKey


# region CreateProcessInstanceByKeyFromStorage
def create_process_instance_by_key_from_storage_example() -> None:
    client = CamundaClient()

    # When restoring a key from a database or message queue,
    # wrap the raw string with the semantic type constructor:
    stored_key = "2251799813685249"  # e.g. from a DB row
    result = client.create_process_instance(
        data=ProcessCreationByKey(
            process_definition_key=ProcessDefinitionKey(stored_key),
        )
    )

    print(f"Process instance key: {result.process_instance_key}")
# endregion CreateProcessInstanceByKeyFromStorage


# region CreateProcessInstanceById
def create_process_instance_by_id_example(process_definition_id: ProcessDefinitionId) -> None:
    client = CamundaClient()

    result = client.create_process_instance(
        data=ProcessCreationById(
            process_definition_id=process_definition_id,
        )
    )

    print(f"Process instance key: {result.process_instance_key}")
# endregion CreateProcessInstanceById


# region CancelProcessInstance
def cancel_process_instance_example(process_definition_id: ProcessDefinitionId) -> None:
    client = CamundaClient()

    # Create a process instance and get its key from the response
    created = client.create_process_instance(
        data=ProcessCreationById(process_definition_id=process_definition_id)
    )

    # Cancel it using the key from the creation response
    client.cancel_process_instance(
        process_instance_key=created.process_instance_key,
    )
# endregion CancelProcessInstance


# region SearchProcessInstances
def search_process_instances_example() -> None:
    client = CamundaClient()

    result = client.search_process_instances(
        data=ProcessInstanceSearchQuery(
            filter_=ProcessInstanceSearchQueryFilter(
                process_definition_id="order-process",
            ),
            sort=[
                ProcessInstanceSearchQuerySortRequest(
                    field=ProcessInstanceSearchQuerySortRequestField.STARTDATE,
                    order=SortOrderEnum.DESC,
                )
            ],
            page=LimitBasedPagination(limit=10),
        )
    )

    for instance in result.items:
        print(f"{instance.process_instance_key}: {instance.state}")
    print(f"Total: {result.page.total_items}")
# endregion SearchProcessInstances

File: .sdk-repos/version-8.9/orchestration-cluster-api-js/examples/process-instance.ts
// Compilable usage examples for process instance operations.
// These examples are type-checked during build to guard against API regressions.

import {
  createCamundaClient,
  type ProcessDefinitionId,
  type ProcessDefinitionKey,
  type ProcessInstanceKey,
} from '@camunda8/orchestration-cluster-api';

//#region CreateProcessInstanceById
async function createProcessInstanceByIdExample(processDefinitionId: ProcessDefinitionId) {
  const camunda = createCamundaClient();

  const result = await camunda.createProcessInstance({
    processDefinitionId,
    variables: {
      orderId: 'ORD-12345',
      amount: 99.95,
    },
  });

  console.log(`Started process instance: ${result.processInstanceKey}`);
}
//#endregion CreateProcessInstanceById

//#region CreateProcessInstanceByKey
async function createProcessInstanceByKeyExample(processDefinitionKey: ProcessDefinitionKey) {
  const camunda = createCamundaClient();

  // Key from a previous API response (e.g. deployment)
  const result = await camunda.createProcessInstance({
    processDefinitionKey,
    variables: {
      orderId: 'ORD-12345',
      amount: 99.95,
    },
  });

  console.log(`Started process instance: ${result.processInstanceKey}`);
}
//#endregion CreateProcessInstanceByKey

//#region CancelProcessInstance
async function cancelProcessInstanceExample(processDefinitionId: ProcessDefinitionId) {
  const camunda = createCamundaClient();

  // Create a process instance and get its key from the response
  const created = await camunda.createProcessInstance({
    processDefinitionId,
  });

  // Cancel the process instance using the key from the creation response
  await camunda.cancelProcessInstance({
    processInstanceKey: created.processInstanceKey,
  });
}
//#endregion CancelProcessInstance

//#region GetProcessInstance
async function getProcessInstanceExample(processInstanceKey: ProcessInstanceKey) {
  const camunda = createCamundaClient();

  const instance = await camunda.getProcessInstance(
    { processInstanceKey },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`State: ${instance.state}`);
  console.log(`Process: ${instance.processDefinitionId}`);
}
//#endregion GetProcessInstance

//#region SearchProcessInstances
async function searchProcessInstancesExample(processDefinitionId: ProcessDefinitionId) {
  const camunda = createCamundaClient();

  const result = await camunda.searchProcessInstances(
    {
      filter: { processDefinitionId },
      sort: [{ field: 'startDate', order: 'DESC' }],
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const instance of result.items ?? []) {
    console.log(`${instance.processInstanceKey}: ${instance.state}`);
  }
  console.log(`Total: ${result.page.totalItems}`);
}
//#endregion SearchProcessInstances

// Suppress "declared but never read"
void createProcessInstanceByIdExample;
void createProcessInstanceByKeyExample;
void cancelProcessInstanceExample;
void getProcessInstanceExample;
void searchProcessInstancesExample;
